### PR TITLE
fix: new exceptions

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change
(A few sentences about this PR)

## Related issues
- Link to issue1 here
- Link to issue1 here

## Checklist for important updates
- [ ] Changelog has been updated
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/pluginInterface/multitenancy/MultitenancyStorage.java [hunk 1]
```diff
@@ -19,21 +19,26 @@
...
-    void addTenantIdInUserPool(TenantIdentifier tenantIdentifier) throws DuplicateTenantException, StorageQueryException;
+    void addTenantIdInUserPool(TenantIdentifier tenantIdentifier)
+            throws DuplicateTenantException, StorageQueryException;
 
     // this also deletes all tenant info from all tables.
     void deleteTenantIdInUserPool(TenantIdentifier tenantIdentifier) throws TenantOrAppNotFoundException;
 
-    void overwriteTenantConfig(TenantConfig config) throws TenantOrAppNotFoundException, StorageQueryException;
+    void overwriteTenantConfig(TenantConfig config) throws TenantOrAppNotFoundException, 
+            DuplicateThirdPartyIdException, DuplicateClientTypeException, StorageQueryException;
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/pluginInterface/multitenancy/MultitenancyStorage.java:41]:
```suggestion
    void overwriteTenantConfig(TenantConfig config) throws TenantOrAppNotFoundException, StorageQueryException;
```

File: src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java [hunk 2]
```diff
@@ -147,7 +147,11 @@ public static class ProviderClient {
         public ProviderClient(@Nullable String clientType, @Nonnull String clientId, @Nullable String clientSecret,
                               @Nullable String[] scope,
                               boolean forcePKCE, @Nullable JsonObject additionalConfig) {
-            this.clientType = clientType;
+            if (clientType == null) {
+                this.clientType = "";
+            } else {
+                this.clientType = clientType;
+            }
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java:154]: if the user sets it to null in the API, it should get returned as null from the API call

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java (hunk 2 above)
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java:154]: no need to do this

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java]: removed. Added check in the postgres plugin.

File: src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java [hunk 3]
```diff
@@ -74,7 +74,6 @@ public static class Provider {
 
         public final boolean requireEmail;
 
-        @Nullable
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/pluginInterface/multitenancy/ThirdPartyConfig.java:77]: @Nonnull

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**